### DISCUSSION TITLE
"Android Remote" - PR request in HA Core

### DISCUSSION THREAD

**Discussion by ChristophCaina** [PRProd/HA-Firemote#61] [General]
I don't know, if you have seen this, but it might be worth to check the current PR for HomeAssistant core:
https://github.com/home-assistant/core/pull/89935

this PR will add a new integration for AndroidTV RemoteController...
maybe, it's worth to check if this integration would allow you to provide a wider range of supported devices; will reduce the amount of work for adding new devices?

**Comment by PRProd** [comment]
WOW!!!!  YES!!!!  This looks like it will revolutionize the ability for me to expand!  I'm going to need to look into this a little bit more.  I was away all day yesterday at the Mayo Clinic, so I'm kind of playing "catch-up" with my regular work duties today.

**Comment by PRProd** [comment]
@moh-Brent - I think you will find this discussion interesting too

**Reply by moh-Brent** [reply]
Game Changer, add BLE support for remotes, save me trying to bodge my way through lol

**Comment by PRProd** [comment]
TODAY IS THE DAY!  Home Assistant 2023.05 has this included now!

I'm happily poking around to see what's possible and what's not.  Sadly, it doesn't seem to like Amazon Fire devices.  The good news is that I should eventually be able to get the NVIDIA Shield buttons to work a million times faster now!  I'm excited!

**Comment by PRProd** [comment]
A user just asked the maintainer of the new Android TV Remote integration about Fire TV Support at https://github.com/home-assistant/core/issues/92656.  I guess now we wait?

In the meantime, I opened my own issue that I found while working on integrating this with Firemote.  https://github.com/home-assistant/core/issues/92658

**Reply by PRProd** [reply]
It was confirmed that Fire TV devices are not supported by the new integration, and likely will not be in the future.  In order for the Android TV Remote integration to work, the device requires https://play.google.com/store/apps/details?id=com.google.android.tv.remote.service to be installed, which isn't natively part of the Fire TV install base.

As far as the one that I opened... the author was a lot smarter than I am when it was designed that way intentionally.  :smile:  I'm updating the beta to accommodate.

**Comment by PRProd** [comment]
# IT'S BETA TESTING TIME EVERYONE!!!

I've just published a new pre-release for everyone to test out - [v2.2.0-b1](https://github.com/PRProd/HA-Firemote/releases/tag/v2.2.0-b1).  Here are the notes I attached to the release:

<hr>

## BETA TIME!
Please note: These improvements WILL NOT WORK with Fire TV Devices.  Also, these speed improvements are so good on the NVIDIA Shield, and the Xiaomi Mi box that you might cry a little tear of joy

Follow a few steps first:
 - KEEP your Android Debug Bridge (formerly known as AndroidTV) integration, do not delete it
 - ADD an Android TV Remote Integration that points to the SAME DEVICE, you will need both for now
 - Download the newest pre-release of Firemote, version 2.2.0-b1
 - Edit a current (or start a new) Firemote card, keeping/adding the Android Debug Bridge Entity (formerly known as the Android TV Entity) and also attach your new Android TV Remote entity under the section "Associated Android TV Remote Entity: (optional)"
 - Hit save, and enjoy the speed!
 - Take notes of any issues you come across

### Known issues:
 - The new Android TV Remote integration does not work with Fire TV Devices right now.  https://github.com/home-assistant/core/issues/92656
 - I updated the way Firemote checks for an active app to use the new Android TV Remote integration instead of using the slower Android Debug bridge integration.  The responsiveness of app buttons lighting up or turning off is a lot better.  However, when you turn your device off, the Android TV Remote integration might still show that there is an app active.  I opened an issue request for the maintainer to look at that.  https://github.com/home-assistant/core/issues/92658

**Follow the Discussion**: https://github.com/PRProd/HA-Firemote/discussions/61

**Full Changelog**: https://github.com/PRProd/HA-Firemote/compare/v2.1.10...v2.2.0-b1

**Reply by moh-Brent** [reply]
it works wickedly fast with Xiaomi Mi Box S/4, you are epic mate :)  No issues that I can find apart from trying to turn the stupid device on over WiFi, still have to use the BT remote for that

**Reply by PRProd** [reply]
@moh-Brent - Right!?  I'm thinking that it's actually faster to respond through Firemote than it is through the physical remote, but that could be because I don't really like the way that remote's buttons feel.

**Comment by PRProd** [comment]
New beta [v2.2.0-b2](https://github.com/PRProd/HA-Firemote/releases/tag/v2.2.0-b2) is available - This version fixes the "lit app button" state when the device is not actually powered on.

**Comment by PRProd** [comment]
[v2.2.0](https://github.com/PRProd/HA-Firemote/releases/tag/v2.2.0) is out of beta now.  :tada:
